WordPress/Classes/ViewRelated/Menus/Views/MenuItemSourceCategoryView.m

```diff
@@ -0,0 +1,173 @@
+    
+    [self showLoadingSourcesIndicatorIfEmpty];
+    void(^stopLoading)() = ^() {
+        [self hideLoadingSourcesIndicator];
+    };
+    PostCategoryService *categoryService = [[PostCategoryService alloc] initWithManagedObjectContext:[self managedObjectContext]];
+    [categoryService syncCategoriesForBlog:[self blog]
+                                    number:@(MenuItemSourceCategorySyncLimit)
+                                    offset:@(0)
+                                   success:^(NSArray<PostCategory *> *categories) {
+                                       stopLoading();
+                                   } failure:^(NSError *error) {
+                                       stopLoading();
+                                       [self showLoadingErrorMessageForResults];
+                                   }];
+}
+
+#pragma mark - TableView methods
+
+- (void)configureSourceCell:(MenuItemSourceCell *)cell forIndexPath:(NSIndexPath *)indexPath
+{
+    PostCategory *category = [self.displayCategories objectAtIndex:indexPath.row];
+    
+    if ([self itemTypeMatchesSourceItemType] && [self.item.contentID integerValue] == [category.categoryID integerValue]) {
+        cell.sourceSelected = YES;
+    } else {
+        cell.sourceSelected = NO;
+    }
+    
+    [cell setTitle:category.categoryName];
+    
+    NSInteger indentationLevel = [[self.categoryIndentationDict objectForKey:[category.categoryID stringValue]] integerValue];
+    [cell setSourceHierarchyIndentation:indentationLevel];
+}
+
+- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
+{
+    [super tableView:tableView didSelectRowAtIndexPath:indexPath];
+    
+    PostCategory *category = [self.displayCategories objectAtIndex:indexPath.row];
+    [self setItemSourceWithContentID:category.categoryID name:category.categoryName];
+    
+    [self deselectVisibleSourceCellsIfNeeded];
+    
+    MenuItemSourceCell *selectedCell = [tableView cellForRowAtIndexPath:indexPath];
+    selectedCell.sourceSelected = YES;
+}
+
+#pragma mark -
+
+- (void)updateDisplayCategories
+{
+    // Get sorted categories by parent/child relationship
+    self.categoryIndentationDict = [NSMutableDictionary dictionary];
+    
+    // Get sorted categories by parent/child relationship
+    WPCategoryTree *tree = [[WPCategoryTree alloc] initWithParent:nil];
+    [tree getChildrenFromObjects:self.resultsController.fetchedObjects];
+    
+    self.displayCategories = [tree getAllObjects];
+    
+    // Get the indentation level of each category.
+    NSUInteger count = [self.displayCategories count];
+    
+    NSMutableDictionary *categoryDict = [NSMutableDictionary dictionary];
+    for (NSInteger i = 0; i < count; i++) {
+        PostCategory *category = [self.displayCategories objectAtIndex:i];
+        [categoryDict setObject:category forKey:category.categoryID];
+    }
+    
+    for (NSInteger i = 0; i < count; i++) {
```

Comment:
Nitpicky. Looks like these `for` loops could be switched to `for..in` syntax to save a couple of lines and avoid the `count` var.